packages/forms/resources/views/components/field-wrapper/label.blade.php

```diff
@@ -13,7 +13,7 @@
 
     <span class="text-sm font-medium leading-6 text-gray-950 dark:text-white">
         {{-- Deliberately poor formatting to ensure that the asterisk sticks to the final word in the label. --}}
-        {{ $slot }}@if ($required && $isMarkedAsRequired && ! $isDisabled)<sup class="text-danger-600 dark:text-danger-400 font-medium">*</sup>
+        {{ $slot }}@if ($required && $isMarkedAsRequired && ! $isDisabled || ! $required && $isMarkedAsRequired && ! $isDisabled)<sup class="text-danger-600 dark:text-danger-400 font-medium">*</sup>
```

Comment:
Isn't that just doing this?

```suggestion
        {{ $slot }}@if (($required || $isMarkedAsRequired) && (! $isDisabled))<sup class="text-danger-600 dark:text-danger-400 font-medium">*</sup>
```